heron/scheduler-core/src/java/com/twitter/heron/scheduler/utils/SchedulerUtils.java

```diff
@@ -39,12 +40,105 @@
 import com.twitter.heron.spi.utils.ShellUtils;
 
 public final class SchedulerUtils {
-  public static final int PORTS_REQUIRED_FOR_EXECUTOR = 9;
+  public static final int PORTS_REQUIRED_FOR_EXECUTOR = ExecutorPorts.getRequiredNumOfPorts();
   public static final int PORTS_REQUIRED_FOR_SCHEDULER = 1;
   public static final String SCHEDULER_COMMAND_LINE_PROPERTIES_OVERRIDE_OPTION = "P";
 
   private static final Logger LOG = Logger.getLogger(SchedulerUtils.class.getName());
 
+  /**
+   * Enum that defines the type of ports that an heron executor needs
+   */
+  public static class ExecutorPorts extends HashMap<ExecutorPorts.ExecutorPortNames, Object> {
```

Comment:
I am also unsure what is the significance of extending a `HashMap` here. Seems unnecessary here.

Reply:
its simpler because if I didn't I would have to have a class variable that is a map which will hold the contents of the portname to port mappings.  I would also need to write wrappers such as "put" to expose the those methods to users or make the variable public.  It the same reason why heron Config extends HashMap as well

Reply:
Based on the current code `put` is the only method that would need to be exposed. As of now other methods from HashMap are not used anywhere. This class is not altering the behavior of the base class either. In which case I would prefer `composition` over `inheritance`